src/platforms/x11/x11_resources.h

```diff
@@ -20,12 +20,19 @@
 #include "mir/geometry/forward.h"
 #include "mir_toolkit/common.h"
 
+#include <stdexcept>
+#include <memory>
+#include <string>
 #include <xcb/xcb.h>
 #include <optional>
 #include <unordered_map>
 #include <functional>
 #include <mutex>
-
+#if __has_include(<format>)
+#include <format>
+#else
+#include <sstream>
+#endif
```

Comment:
I don't think any of these changes are needed in _this header_. Those additional headers needed by the implementation should be added to the implementation file.